### PR TITLE
Node notes cli

### PR DESCRIPTION
**Pull request by brian-luong**
gif
screenshot

### CODE REVIEW

File: node-notes-cli/app.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+if (process.argv[2] === 'read') {
+  for (const property in dataJson.notes) {
+    console.log(`${property}: ${dataJson.notes[property]}`);
+  }
+} else if (process.argv[2] === 'create' && process.argv[3]) {
+  dataJson.notes[nextId] = process.argv[3];
+  dataJson.nextId++;
+} else if (process.argv[2] === 'update' && process.argv[3] < nextId && process.argv[4]) {
+  dataJson.notes[process.argv[3]] = process.argv[4];
```
Comment (TimelikeClosure) [inline node-notes-cli/app.js]: The `update` operation should only be able to modify existing notes, not create new ones. Keep in mind that, in addition to the user being able to supply invalid `id`s, the `delete` operation can remove notes.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: node-notes-cli/app.js (hunk 1 above)
Comment (brian-luong) [reply node-notes-cli/app.js]: hi @TimelikeClosure, I updated it so "update" can only modify existing ones. Could you help clarify "user being able to supply invalid ids?" The delete operation should remove notes, please let me know if I'm missing anything! Thanks

**Review by TimelikeClosure — APPROVED** [review]

File: node-notes-cli/app.js (hunk 1 above)
Comment (TimelikeClosure) [reply node-notes-cli/app.js]: @brian-luong That means if the user does something like this, the note `5` should stay deleted:

```bash
node app.js delete 5
node app.js update 5 "back from the dead"
```

Comment (brian-luong) [reply node-notes-cli/app.js]: Oh got it, thank you

**Pull request merged**